src/tty/mod.rs

```diff
@@ -53,6 +59,29 @@ pub trait EventedReadWrite {
     fn write_token(&self) -> mio::Token;
 }
 
+/// Events concerning TTY child processes
+#[derive(PartialEq)]
+pub enum ChildEvent {
+    /// Indicates the child has exited
+    Exited
+}
+
+/// A pseudoterminal (or PTY)
+///
+/// This is a refinement of EventedReadWrite that also provides a channel through which we can be
+/// notified if the PTY child process does something we care about (other than writing to the TTY).
+/// In particular, this allows for race-free child exit notification on UNIX (cf. `SIGCHLD`).
+pub trait EventedPty : EventedReadWrite {
+    #[cfg(unix)]
+    fn child_event_token(&self) -> mio::Token;
+
+    /// Tries to retrieve an event
+    ///
+    /// Returns `Some(event)` on success, or `None` if there are no events to retrieve.
+    #[cfg(unix)]
+    fn get_child_event(&mut self) -> Option<ChildEvent>;
```

Comment:
`get_child_event` sounds like it just retrieves a field variable (like a getter), however if I understand things correctly, this is more like pulling the next event out of the PTY, right? It should probably be called something like `next_child_event` then, to represent that calling it twice in quick succession usually never returns the same value?

Reply:
Agreed - done.